packages/babel-plugin-transform-inline-consecutive-adds/__tests__/transform-inline-consecutive-adds-test.js

```diff
@@ -0,0 +1,225 @@
+  it("should collapse simple consecutive", () => {
+    const source = unpad(`
+      const foo = {
+        z: 3.0
+      };
+      foo.a = 42;
+      foo.b = ["hi"];
+      foo.c = bar();
+      foo.d = "str";
+    `);
+    const expected = unpad(`
+      const foo = {
+        z: 3.0,
+        a: 42,
+        b: ["hi"],
+        c: bar(),
+        d: "str"
+      };
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should collapse only up to last ExpressionStatement", () => {
+    const source = unpad(`
+      const foo = {};
+      foo.a = 42;
+      console.log(foo);
+    `);
+    const expected = unpad(`
+      const foo = {
+        a: 42
+      };
+
+      console.log(foo);
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should not collapse if lval is nested MemberExpression", () => {
+    const source = unpad(`
+      const foo = {};
+      foo.bar.a = 42;
+    `);
+    expect(transform(source)).toBe(source);
+  });
+
+  it("should not collapse if lval has wrong name", () => {
+    const source = unpad(`
+      const foo = {};
+      bar.a = 42;
+    `);
+    expect(transform(source)).toBe(source);
+  });
+
+  it("should not collapse if has direct dependency issues", () => {
+    const source = unpad(`
+      const foo = {};
+      foo.a = function () {
+        console.log(3);
+      };
+      foo.b = foo.a();
+    `);
+    const expected = unpad(`
+      const foo = {
+        a: function () {
+          console.log(3);
+        }
+      };
+
+      foo.b = foo.a();
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should not collapse if has indirect dependency issues", () => {
+    const source = unpad(`
+      var foo = {};
+      foo.a = 4;
+      foo.b = cat();
+      function cat() {
+        return bar();
+      }
+      function bar() {
+        console.log(foo);
+        return 0;
+      }
+    `);
+    const expected = unpad(`
+      var foo = {
+        a: 4
+      };
+
+      foo.b = cat();
+      function cat() {
+        return bar();
+      }
+      function bar() {
+        console.log(foo);
+        return 0;
+      }
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+
+  it("should not collapse if has indirect, nested dependency issues", () => {
+    const source = unpad(`
+      var foo = {};
+      foo.a = 4;
+      foo.b = cat();
+      function cat() {
+        return bar();
+        function bar() {
+          console.log(foo);
+          return 0;
+        }
+      }
+    `);
+    const expected = unpad(`
+      var foo = {
+        a: 4
+      };
+
+      foo.b = cat();
+      function cat() {
+        return bar();
+        function bar() {
+          console.log(foo);
+          return 0;
+        }
+      }
+    `);
+    expect(transform(source)).toBe(expected);
+  });
+
+  it("should collapse normal computed properties", () => {
+    const source = unpad(`
+      var foo = {};
+      foo["a"] = 0;
```

Comment:
These are not computed properties though. It's just a regular bracket-based property resolution. What I wanted us to do is transform **to** computed property, when there's an identifier `var foo = {}; foo[x] = 1` to `var foo = { [x]: 1 };`.

However, on a second thought, this would mean we're turning potentially ES5 code to ES6 so it's a special kind of transformation. Let's ignore that for now and revisit later.